### ISSUE TITLE
Way to specify .nupkg file name?

### ISSUE THREAD

**Issue by jonathanpeppers** [NuGet/Home#10834]
## Details about Problem

It appears that NuGet creates a file name for `.nupkg` files such as:

    $(PackageId).$(PackageVersion).nupkg

Where this gets weird, is with the [.NET Workload Spec](https://github.com/dotnet/designs/pull/188/files#diff-8fcaa29d8e6f00b34b3cb1830d93f33e75f04424780a66a3c658c7021048e74fR125).

If we need to make a package with:

* `PackageId`: `Microsoft.NET.Sdk.Android.Manifest-6.0.200`
* `PackageVersion`: `11.0.200-preview.4.245`

We end up with the file named: `Microsoft.NET.Sdk.Android.Manifest-6.0.200.11.0.200-preview.4.245.nupkg`

Is there a way to put a `-` in the file name instead of a `.`? Or just an MSBuild property to set any custom file name?

If not, consider this a "feature request". Thanks!

## Repro steps and/or sample project

I think you could just `pack` a new `.csproj` with `$(PackageId)` and `$(PackageVersion)` as in the example above.

**Comment by erdembayar** [comment]
@jonathanpeppers 
Thank you for filing this issue. 
I tried your scenario and here is solution.
1. mkdir Microsoft.NET.Sdk.Android.Manifest-6.0.200
2. cd Microsoft.NET.Sdk.Android.Manifest-6.0.200
3. dotnet new classlib
4. msbuild -t:pack
![image](https://user-images.githubusercontent.com/8766776/117893472-56a99380-b26f-11eb-8ea4-a7d234794173.png)

5. msbuild -t:pack -p:PackageId=Microsoft.NET.Sdk.Android.Manifest.6.0.200 -p:packageversion=11.0.200-preview.4.245
I believe below is what you wanted, right?
![image](https://user-images.githubusercontent.com/8766776/117893643-a720f100-b26f-11eb-8299-aba7e075a870.png)

Please let me know if this resolves your problem.

**Comment by jonathanpeppers** [comment]
@erdembayar in your screenshot, we want there to be a `-` between `6.0.200` and `11.0.200` *instead* of a `.`. One is coming from the `$(PackageId)` and the other is the `$(PackageVersion)`.

Also assume someone might want `foo.nupkg` for this example, can there be a completely custom file name?

**Comment by erdembayar** [comment]
@jonathanpeppers 
My test shows you can name your nupkg with any arbitrary name, so make it easy instead of 2 parameters just pass 1(name+version).
`msbuild -t:pack -p:PackageId=Foo11.0.200-preview.4.245`

![image](https://user-images.githubusercontent.com/8766776/118048737-8f11a600-b331-11eb-97b1-46d2dbaaa8a7.png)

**Comment by jonathanpeppers** [comment]
Right, but we're asking to change the name of the file independent of the `$(PackageId)` and `$(PackageVersion)`, is there a way to do that?

**Comment by nkolev92** [comment]
@jonathanpeppers There's no way to do that today. 

A recommendation we can make with the current tooling is to rename the file after pack.
